helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -80,7 +82,19 @@
     DISABLED_INSTANCES,
 
     // Specifies job types and used for quota allocation
-    QUOTA_TYPES
+    QUOTA_TYPES,
+
+    // The required instance capacity keys for resource partition assignment calculation.
+    INSTANCE_CAPACITY_KEYS,
+    // The preference of the rebalance result.
+    // EVENNESS - Evenness of the resource utilization, partition, and top state distribution.
+    // LESS_MOVEMENT - the tendency of keeping the current assignment instead of moving the partition for optimal assignment.
+    REBALANCE_PREFERENCE
+  }
+
+  public enum GlobalRebalancePreferenceKey {
```

Comment:
Will it be clearer if you combine all the rebalance related configuration in a separate class or internal class of ClusterConfig?

Reply:
Separate class implies a separate config file. Which we don't want.
As for the internal class, since we only have 2 items, It seems to be overhead for me. Note that how we organize the class has no backward-compatible concern. So if we have more items later, we can split.